Server/JokesOnYou.Web.Api/Repositories/JokesRepository.cs

```diff
@@ -0,0 +1,43 @@
+﻿using AutoMapper;
+using AutoMapper.QueryableExtensions;
+using JokesOnYou.Web.Api.Data;
+using JokesOnYou.Web.Api.DTOs;
+using JokesOnYou.Web.Api.Models;
+using JokesOnYou.Web.Api.Repositories.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace JokesOnYou.Web.Api.Repositories
+{
+    public class JokesRepository : IJokesRepository
+    {
+        private readonly DataContext _context;
+
+        public JokesRepository(DataContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<IEnumerable<Joke>> GetAllJokesAsync()
+        {
+            var jokes = await _context.Jokes.Include( joke => joke.Author).ToListAsync();
+            return jokes;
+        }
+
+        public async Task<IEnumerable<JokeReplyDto>> GetAllJokeDtosAsync()
+        {
+            var configuration = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<Joke, JokeReplyDto>();
+                cfg.CreateMap<User, JokeAuthorDto>();
```

Comment:
Any other way of doing this since i already have the mapping profile.

Reply:
Yes. So since we have the profile configured, and automapper is added to DI serviceprovider. We can just inject the mapper using IMapper interface and use it all over the place.

Reply:
Very nice! 
_mapper.ConfigurationProvider worked here I was thinking there should be a way :D